pkg/controller/jobframework/integrationmanager.go

```diff
@@ -53,6 +53,8 @@ type ReconcilerFactory func(client client.Client, record record.EventRecorder, o
 
 // IntegrationCallbacks groups a set of callbacks used to integrate a new framework.
 type IntegrationCallbacks struct {
+	// NewJob creates a new instance of job
+	NewJob func() GenericJob
```

Comment:
@alculquicondor I added this field as required. Do we have some docs where mentioned which field is required?

Reply:
We could enhance this document

https://kueue.sigs.k8s.io/docs/tasks/dev/integrate_a_custom_job/

Reply:
Yes, I think it would be great.